src/psyclone/psyir/nodes/node.py

```diff
@@ -1284,6 +1327,30 @@ def preceding(self, reverse=False, routine=True):
             nodes.reverse()
         return nodes
 
+    def immediatelyPrecedes(self, node_2):
```

Comment:
Happy to change name to `immediately_precedes` if preferred (I was following `sameParent`).